test/Autofac.Test/Features/LightweightAdapters/LightweightAdapterRegistrationExtensionsTests.cs

```diff
@@ -81,6 +81,34 @@ public void AdaptedMetadataIsPassed()
             }
         }
 
+        public class DecoratingServiceThatHasDefaultImplementation
+        {
+            readonly IContainer _container;
+
+            public DecoratingServiceThatHasDefaultImplementation()
+            {
+                const string from = "from";
+                var builder = new ContainerBuilder();
+
+                // Implementer1 is the default implementation for IService
+                builder.RegisterType<Implementer1>().As<IService>();
```

Comment:
@tillig yes, it still fails. I wanted to keep it more closer to real-world usage. Updated the commit.